{"event": "config", "data": {"system": {"arch": "cuda", "sshkey": null, "nodes": [{"name": "local", "ip": "127.0.0.1", "sshport": 22, "user": "root", "main": true, "hostname": "cn-d001.server.mila.quebec", "local": true}], "self": {"name": "local", "ip": "127.0.0.1", "sshport": 22, "user": "root", "main": true, "hostname": "cn-d001.server.mila.quebec", "local": true}}, "dirs": {"base": "/network/scratch/o/ortizgas/data/milabench", "venv": "/network/scratch/o/ortizgas/data/milabench/venv/torch", "data": "/network/scratch/o/ortizgas/data/milabench/data", "runs": "/network/scratch/o/ortizgas/data/milabench/runs", "extra": "/network/scratch/o/ortizgas/data/milabench/extra/rlhf-gpus", "cache": "/network/scratch/o/ortizgas/data/milabench/cache"}, "group": "rlhf-gpus", "install_group": "torch", "install_variant": "cuda", "run_name": "NVIDIA-A100-SXM4-40GB_rlhf-gpus_256.staging", "enabled": true, "capabilities": {"nodes": 1}, "max_duration": 600, "voir": {"options": {"stop": 60, "interval": "1s"}}, "validation": {"usage": {"gpu_load_threshold": 0.5, "gpu_mem_threshold": 0.5}}, "config_base": "/home/mila/o/ortizgas/CODE/milabench/config", "config_file": "/home/mila/o/ortizgas/CODE/milabench/config/standard.yaml", "definition": "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf", "plan": {"method": "njobs", "n": 1}, "tags": ["llm", "multigpu", "rl", "rlhf"], "argv": {"--output_dir": "{milabench_extra}/output", "--model_name_or_path": "EleutherAI/pythia-1b-deduped", "--per_device_train_batch_size": 64, "--logging_strategy": "no", "--log_level": "critical", "--bf16": true}, "weight": 1.0, "name": "rlhf-gpus", "tag": ["rlhf-gpus", "0"], "job-number": 0, "devices": [0, 1, 2, 3]}, "pipe": null}
{"event": "meta", "data": {"cpu": {"count": 256, "brand": "AMD EPYC 7742 64-Core Processor"}, "os": {"sysname": "Linux", "nodename": "cn-d001.server.mila.quebec", "release": "5.15.0-1048-nvidia", "version": "#48-Ubuntu SMP Thu Mar 21 18:19:02 UTC 2024", "machine": "x86_64"}, "accelerators": {"arch": "cuda", "gpus": {"0": {"minor_number": "1", "device": 0, "product": "NVIDIA A100-SXM4-40GB", "memory": {"used": 620.875, "total": 40960.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 27.0, "power": 55.396, "selection_variable": "CUDA_VISIBLE_DEVICES"}, "1": {"minor_number": "2", "device": 1, "product": "NVIDIA A100-SXM4-40GB", "memory": {"used": 620.875, "total": 40960.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 27.0, "power": 52.063, "selection_variable": "CUDA_VISIBLE_DEVICES"}, "2": {"minor_number": "3", "device": 2, "product": "NVIDIA A100-SXM4-40GB", "memory": {"used": 620.875, "total": 40960.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 27.0, "power": 56.202, "selection_variable": "CUDA_VISIBLE_DEVICES"}, "3": {"minor_number": "7", "device": 3, "product": "NVIDIA A100-SXM4-40GB", "memory": {"used": 620.875, "total": 40960.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 31.0, "power": 55.06, "selection_variable": "CUDA_VISIBLE_DEVICES"}}}, "date": 1733964954.396825, "milabench": {"tag": "v1.0.0_RC1-13-g7a90b16", "commit": "7a90b1691650232ecd63abded0b7be84bb294c05", "date": "2024-11-08 11:28:56 -0500"}, "pytorch": {"torch": "2.4.0+cu121", "compiler": "GCC 9.3", "cpp": "C++ Version: 201703", "intel": "Intel(R) oneAPI Math Kernel Library Version 2022.2-Product Build 20220804 for Intel(R) 64 architecture applications", "mkl": "OpenMP 201511 (a.k.a. OpenMP 4.5)", "openmp": "OpenMP 201511 (a.k.a. OpenMP 4.5)", "lapack": "LAPACK is enabled (usually provided by MKL)", "nnpack": "NNPACK is enabled", "cpu": "CPU capability usage: AVX2", "build_settings": {"BLAS_INFO": "mkl", "BUILD_TYPE": "Release", "CUDA_VERSION": "12.1", "CUDNN_VERSION": "9.1.0", "CXX_COMPILER": "/opt/rh/devtoolset-9/root/usr/bin/c++", "CXX_FLAGS": "-D_GLIBCXX_USE_CXX11_ABI=0 -fabi-version=11 -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-stringop-overflow -Wsuggest-override -Wno-psabi -Wno-error=pedantic -Wno-error=old-style-cast -Wno-missing-braces -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow", "LAPACK_INFO": "mkl", "PERF_WITH_AVX": "1", "PERF_WITH_AVX2": "1", "PERF_WITH_AVX512": "1", "TORCH_VERSION": "2.4.0", "USE_CUDA": "ON", "USE_CUDNN": "ON", "USE_CUSPARSELT": "1", "USE_EXCEPTION_PTR": "1", "USE_GFLAGS": "OFF", "USE_GLOG": "OFF", "USE_GLOO": "ON", "USE_MKL": "ON", "USE_MKLDNN": "ON", "USE_MPI": "OFF", "USE_NCCL": "1", "USE_NNPACK": "ON", "USE_OPENMP": "ON", "USE_ROCM": "OFF", "USE_ROCM_KERNEL_ASSERT": "OFF"}}}, "pipe": null}
{"event": "start", "data": {"command": ["/home/mila/o/ortizgas/CODE/milabench/milabench/scripts/activator", "/network/scratch/o/ortizgas/data/milabench/venv/torch", "/network/scratch/o/ortizgas/data/milabench/cache", "accelerate", "launch", "--mixed_precision=bf16", "--dynamo_backend=no", "--machine_rank=0", "--num_machines=1", "--multi_gpu", "--gradient_accumulation_steps=1", "--num_cpu_threads_per_process=4", "--main_process_ip=127.0.0.1", "--main_process_port=29400", "--num_processes=4", "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py", "--output_dir", "/network/scratch/o/ortizgas/data/milabench/extra/rlhf-gpus/output", "--model_name_or_path", "EleutherAI/pythia-1b-deduped", "--per_device_train_batch_size", "256", "--logging_strategy", "no", "--log_level", "critical", "--bf16"], "time": 1733946954.5205607}, "pipe": null}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/cache\n", "pipe": "stdout"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "data", "data": {"task": "main", "time": 1733946982.9528966, "gpudata": {"0": {"memory": [2055.6875, 40960.0], "load": 0.0, "temperature": 26.0, "power": 63.57}, "1": {"memory": [2199.6875, 40960.0], "load": 0.0, "temperature": 26.0, "power": 60.565}, "2": {"memory": [2199.6875, 40960.0], "load": 0.0, "temperature": 27.0, "power": 63.776}, "3": {"memory": [2055.6875, 40960.0], "load": 0.0, "temperature": 31.0, "power": 63.226}}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733946982.9528966, "iodata": {"read_count": 78, "write_count": 788, "read_bytes": 211968, "read_time": 3, "write_time": 52, "busy_time": 136}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733946982.9528966, "netdata": {"bytes_sent": 3479433316600, "bytes_recv": 4715774313306, "packets_sent": 2869967124, "packets_recv": 4047833389, "errin": 0, "errout": 0, "dropin": 2022, "dropout": 0}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733946982.9528966, "cpudata": {"memory": [58502434816, 1081172426752], "load": 3.4}}, "pipe": "data"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "line", "data": "===training policy===\n", "pipe": "stdout"}
{"event": "line", "data": "\r  0%|          | 0/16 [00:00<?, ?it/s]", "pipe": "stderr"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "data", "data": {"progress": [0, 70], "task": "early_stop"}, "pipe": "data"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "data", "data": {"task": "main", "time": 1733946993.0792596, "gpudata": {"0": {"memory": [8377.6875, 40960.0], "load": 0.82, "temperature": 32.0, "power": 92.843}, "1": {"memory": [12975.6875, 40960.0], "load": 0.72, "temperature": 36.0, "power": 322.177}, "2": {"memory": [11235.6875, 40960.0], "load": 0.79, "temperature": 33.0, "power": 170.002}, "3": {"memory": [10001.6875, 40960.0], "load": 0.68, "temperature": 37.0, "power": 182.555}}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733946993.0792596, "iodata": {"read_count": 90, "write_count": 921, "read_bytes": 269312, "read_time": 5, "write_time": 55, "busy_time": 192}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733946993.0792596, "netdata": {"bytes_sent": 3479434067153, "bytes_recv": 4715774985875, "packets_sent": 2869967742, "packets_recv": 4047833971, "errin": 0, "errout": 0, "dropin": 2022, "dropout": 0}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733946993.0792596, "cpudata": {"memory": [50081034240, 1081172426752], "load": 4.2}}, "pipe": "data"}
{"event": "line", "data": "[rank1]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 439, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     new_all_logprobs = F.log_softmax(logits, dim=-1)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/nn/functional.py\", line 1977, in log_softmax\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     ret = input.log_softmax(dim)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 2.54 GiB. GPU 1 has a total capacity of 39.39 GiB of which 554.31 MiB is free. Including non-PyTorch memory, this process has 38.84 GiB memory in use. Of the allocated memory 36.73 GiB is allocated by PyTorch, and 502.90 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 439, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     new_all_logprobs = F.log_softmax(logits, dim=-1)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/nn/functional.py\", line 1977, in log_softmax\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     ret = input.log_softmax(dim)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 2.54 GiB. GPU 0 has a total capacity of 39.39 GiB of which 1.03 GiB is free. Including non-PyTorch memory, this process has 38.36 GiB memory in use. Of the allocated memory 36.51 GiB is allocated by PyTorch, and 372.64 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 439, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     new_all_logprobs = F.log_softmax(logits, dim=-1)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/nn/functional.py\", line 1977, in log_softmax\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     ret = input.log_softmax(dim)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 2.54 GiB. GPU 3 has a total capacity of 39.39 GiB of which 1.33 GiB is free. Including non-PyTorch memory, this process has 38.05 GiB memory in use. Of the allocated memory 36.00 GiB is allocated by PyTorch, and 589.33 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 439, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     new_all_logprobs = F.log_softmax(logits, dim=-1)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/nn/functional.py\", line 1977, in log_softmax\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     ret = input.log_softmax(dim)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 2.54 GiB. GPU 2 has a total capacity of 39.39 GiB of which 1.01 GiB is free. Including non-PyTorch memory, this process has 38.37 GiB memory in use. Of the allocated memory 36.24 GiB is allocated by PyTorch, and 521.01 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "\r  0%|          | 0/16 [00:11<?, ?it/s]\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:[W1211 14:56:36.231659434 ProcessGroupNCCL.cpp:1168] Warning: WARNING: process group has NOT been destroyed before we destruct ProcessGroupNCCL. On normal program exit, the application should call destroy_process_group to ensure that any pending NCCL operations have finished in this process. In rare cases this process can exit before this point and block the progress of another member of the process group. This constraint has always been present,  but this warning has only been added since PyTorch 2.4 (function operator())\n", "pipe": "stderr"}
{"event": "line", "data": "W1211 14:56:38.253396 139979227411584 torch/distributed/elastic/multiprocessing/api.py:858] Sending process 2838602 closing signal SIGTERM\n", "pipe": "stderr"}
{"event": "line", "data": "W1211 14:56:38.253757 139979227411584 torch/distributed/elastic/multiprocessing/api.py:858] Sending process 2838603 closing signal SIGTERM\n", "pipe": "stderr"}
{"event": "line", "data": "W1211 14:56:38.253824 139979227411584 torch/distributed/elastic/multiprocessing/api.py:858] Sending process 2838604 closing signal SIGTERM\n", "pipe": "stderr"}
{"event": "line", "data": "E1211 14:56:39.182284 139979227411584 torch/distributed/elastic/multiprocessing/api.py:833] failed (exitcode: 1) local_rank: 3 (pid: 2838605) of binary: /network/scratch/o/ortizgas/data/milabench/venv/torch/bin/python\n", "pipe": "stderr"}
{"event": "line", "data": "Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/bin/accelerate\", line 8, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "    sys.exit(main())\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/commands/accelerate_cli.py\", line 48, in main\n", "pipe": "stderr"}
{"event": "line", "data": "    args.func(args)\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/commands/launch.py\", line 1165, in launch_command\n", "pipe": "stderr"}
{"event": "line", "data": "    multi_gpu_launcher(args)\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/commands/launch.py\", line 799, in multi_gpu_launcher\n", "pipe": "stderr"}
{"event": "line", "data": "    distrib_run.run(args)\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/distributed/run.py\", line 892, in run\n", "pipe": "stderr"}
{"event": "line", "data": "    elastic_launch(\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/distributed/launcher/api.py\", line 133, in __call__\n", "pipe": "stderr"}
{"event": "line", "data": "    return launch_agent(self._config, self._entrypoint, list(args))\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/distributed/launcher/api.py\", line 264, in launch_agent\n", "pipe": "stderr"}
{"event": "line", "data": "    raise ChildFailedError(\n", "pipe": "stderr"}
{"event": "line", "data": "torch.distributed.elastic.multiprocessing.errors.ChildFailedError: \n", "pipe": "stderr"}
{"event": "line", "data": "============================================================\n", "pipe": "stderr"}
{"event": "line", "data": "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py FAILED\n", "pipe": "stderr"}
{"event": "line", "data": "------------------------------------------------------------\n", "pipe": "stderr"}
{"event": "line", "data": "Failures:\n", "pipe": "stderr"}
{"event": "line", "data": "  <NO_OTHER_FAILURES>\n", "pipe": "stderr"}
{"event": "line", "data": "------------------------------------------------------------\n", "pipe": "stderr"}
{"event": "line", "data": "Root Cause (first observed failure):\n", "pipe": "stderr"}
{"event": "line", "data": "[0]:\n", "pipe": "stderr"}
{"event": "line", "data": "  time      : 2024-12-11_14:56:38\n", "pipe": "stderr"}
{"event": "line", "data": "  host      : cn-d001.server.mila.quebec\n", "pipe": "stderr"}
{"event": "line", "data": "  rank      : 3 (local_rank: 3)\n", "pipe": "stderr"}
{"event": "line", "data": "  exitcode  : 1 (pid: 2838605)\n", "pipe": "stderr"}
{"event": "line", "data": "  error_file: <N/A>\n", "pipe": "stderr"}
{"event": "line", "data": "  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html\n", "pipe": "stderr"}
{"event": "line", "data": "============================================================\n", "pipe": "stderr"}
{"event": "end", "data": {"command": ["/home/mila/o/ortizgas/CODE/milabench/milabench/scripts/activator", "/network/scratch/o/ortizgas/data/milabench/venv/torch", "/network/scratch/o/ortizgas/data/milabench/cache", "accelerate", "launch", "--mixed_precision=bf16", "--dynamo_backend=no", "--machine_rank=0", "--num_machines=1", "--multi_gpu", "--gradient_accumulation_steps=1", "--num_cpu_threads_per_process=4", "--main_process_ip=127.0.0.1", "--main_process_port=29400", "--num_processes=4", "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py", "--output_dir", "/network/scratch/o/ortizgas/data/milabench/extra/rlhf-gpus/output", "--model_name_or_path", "EleutherAI/pythia-1b-deduped", "--per_device_train_batch_size", "256", "--logging_strategy", "no", "--log_level", "critical", "--bf16"], "time": 1733946999.815175, "return_code": 1}, "pipe": null}
